src/AppInstallerCLICore/Commands/UpgradeCommand.cpp

```diff
@@ -98,6 +98,9 @@ namespace AppInstaller::CLI
             Argument::ForType(Args::Type::Log),             // -o
             Argument::ForType(Args::Type::Override),
             Argument::ForType(Args::Type::InstallLocation), // -l
+            // Alias 'a' already used by --all, so use alternative name "ua" here

```

Comment:
Just putting this out there - `-a` for `--all` hasn't appeared in any releases yet, so if @denelon thinks that it would be best to keep `-a` standard to `--architecture` then it would probably be best to make that change now rather than later. 

`-*` could be an option, so could `-+`

Reply:
@denelon here is what the discussion is.

Issue: we are adding --architecture arg to upgrade command, --architecture args has alias -a in Install command, but -a is used by --all in upgrade command(this has not been officially released yet), do you think we should change --all alias to another one and make -a the alias of --architecture that's consistent for both Install command and Upgrade command?

I'm leaning towards @Trenly 's suggestion. What do you think?

Reply:
We shouldn't break any other existing arguments that work. If necessary, we could look at all the arguments against the 2.0 breaking changes Issue and consider some new aliases. I could see "x" for architecture since most of them are x86, x64, etc... but I want to be careful about an argument's alias on one command having a different meaning on a different command.

Reply:
Not that it matters here, but another thing to consider is that `winget show` and `winget list` don't currently support the `--architecture` argument. 

I think keeping `-a` for `--architecture` makes the most sense for now. As the upgrade command becomes more complex, I'm sure that flags for `--include-pinned` and `--include-explicit` will be wanted, so it may pay to pre-think of what some of those aliases may be

Reply:
To be clear, -a for --all in upgrade command is not released in any of the releases yet so it's Not a breaking change to remove -a as alias for --all and give -a back to --architecture.

Reply:
Had quick sync with Demitrius, we are fine with reverting previous change and give -a back to --architecture to be consistent

Reply:
@Trenly we should think up a reasonable alias for --all "-e" for everything lol!

Reply:
Either that or start using emoji

Reply:
> @Trenly we should think up a reasonable alias for --all "-e" for everything lol!

🤕 -e, --exact

These are the ones currently not in use for upgrade - `b, c, d, f, g, j, k, n, p, r, t, u, w, x, y, z`

I'd propose:
```yaml
-r : --all # --recurse as Alternate Name
-x : --include-excluded # once winget exclude is added
-y : --include-explicit # to allow for upgrading of packages where RequireExplicitUpgrade with --all
-u : --include-unknown 
-p : --include-pinned # once package pinning is implemented
```
That would allow users to do `winget upgrade -ru` to upgrade all packages including those with unknown version of `winget upgrade -yupr` to upgrade everything except excluded packages